forward_engineering/service/elastic_search_service/client_factory.js

```diff
@@ -0,0 +1,153 @@
+const {ConnectionType} = require("../../../enums/connection_type_enum");
+const fs = require('fs');
+const elasticsearch = require('@elastic/elasticsearch');
+const {assertExists} = require("../../../util/error-util");
+
+class ElasticSearchClientFactory {
+
+    /**
+     * @param protocol {string}
+     * @param host {string}
+     * @param port {number | string}
+     * @param path {string | undefined}
+     * @return {URL}
+     * @throws {Error}
+     * */
+    static #buildUrl(protocol, host, port, path) {
+        assertExists(protocol, 'protocol');
+        assertExists(host, 'host');
+        assertExists(port, 'port');
+        const pathAsString = path ? String(path) : '';
+        const pathNoStartSlash = pathAsString.startsWith('/') ? pathAsString.substring(1) : pathAsString;
+        const urlAsString = `${protocol}://${host}:${port}/${pathNoStartSlash}`;
+        return new URL(urlAsString);
+    }
+
+    /**
+     * @param connectionInfo {{
+     *     connectionType: string,
+     *     username: string,
+     *     password: string,
+     *     protocol: string,
+     *     host?: string,
+     *     port?: number,
+     *     path?: string,
+     *     hosts?: Array<{
+     *         host: string,
+     *         port: string,
+     *     }>,
+     *     is_ssl: boolean,
+     *     ca?: string,
+     *     rejectUnauthorized?: boolean,
+     * }}
+     * @throws Error
+     * */
+    static #applyDefaults(connectionInfo) {
+        connectionInfo.protocol = connectionInfo.protocol || 'http';
+        connectionInfo.connectionType = connectionInfo.connectionType || ConnectionType.DIRECT_CONNECTION;
+    }
+
+    /**
+     * @param connectionInfo {{
+     *     connectionType: string,
+     *     username: string,
+     *     password: string,
+     *     protocol: string,
+     *     host?: string,
+     *     port?: number,
+     *     path?: string,
+     *     hosts?: Array<{
+     *         host: string,
+     *         port: string,
+     *     }>,
+     *     is_ssl: boolean,
+     *     ca?: string,
+     *     rejectUnauthorized?: boolean,
+     * }}
+     * @throws Error
+     * */
+    static getByConnectionInfo(connectionInfo) {
```

Comment:
Probably better to store connection and when already exist established connection use him instead of create new connection each time